FIX count wall time instead of CPU time

wiiha opened the pull request:
Intervall only accounts for CPU time
passed, not actual wall time passed
resulting in the timer going slower
than actual wall time.

CPU time vs Wall time
When a browser tab or window is off-screen
or in a tab that isn't focused, the scheduler
doesn't execute the JS engine every
tick/second. Instead, the process is scheduled
to run in less frequent intervals. This means
that one can't assume that the interval timer
will execute once every wall time second, but
rather once every "JS engine active" second.
Since the countdown should count wall time,
we need to calculate a timestamp based delta
for when the function last ran.

wiiha committed:
FIX count wall time instead of CPU time

Intervall only accounts for CPU time
passed, not actual wall time passed.

CPU time vs Wall time
When a browser tab or window is off-screen
or in a tab that isn't focused, the scheduler
doesn't execute the JS engine every
tick/second. Instead, the process is scheduled
to run in less frequent intervals. This means
that one can't assume that the interval timer
will execute once every wall time second, but
rather once every "JS engine active" second.
Since the countdown should count wall time,
we need to calculate a delta for when the
function last ran.

wiiha commented:
@agustinl, if you would like me to open the PR against another branch feel free to write that in a comment. 👍
I have used this almost daily since you presented it on HN. It's a very nice experience.

agustinl commented:
Hi @wiiha ! Thanks for your time to this PR!

I found an issue: when you pause the pomodoro timer, its still running in background, so when you play again, they discount the seconds that it was paused.

On the other hand, do you have any example of how to replicate this issue?

wiiha committed:
FIX paused time no longer ignored

Time passing while paused was accounted for
when the timer was again started. It should
not be the case.

wiiha commented:
Nice catch @agustinl, I fixed the pause now.
Regarding replicating the issue. Execute the following code snippet in the dev tools console and then move to another tab (or let the tab be off screen) for say 5 minutes. When you come back you should see a difference in the time output.

```javascript

var t0 = Date.now();
var intervallSeconds = 1;
setInterval(()=>{
   console.log({intervallSeconds,timestampSeconds:Math.round((Date.now()-t0)/1000)})
    intervallSeconds = intervallSeconds + 1
},1000)

```

**Before switching tab**
<img width="621" alt="image" src="https://user-images.githubusercontent.com/26814716/224484196-7a429dcd-745f-4d97-8463-73a5dadb45f6.png">

**After switching tab and coming back**
<img width="627" alt="image" src="https://user-images.githubusercontent.com/26814716/224484277-f8ae9bc4-9a49-485c-8596-1d35a68a1d12.png">

agustinl commented:
Works perfectly @wiiha 🙌

Merge to release **v4.0.0**

Thanks for your support!
